google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreOptions.java

```diff
@@ -69,6 +75,10 @@ public ServiceRpc create(DatastoreOptions options) {
         if (options.getTransportOptions() instanceof GrpcTransportOptions) {
           return new GrpcDatastoreRpc(options);
         } else if (options.getTransportOptions() instanceof HttpTransportOptions) {
+          // todo see if we can remove this check
+          if (DatastoreUtils.isEmulator(options)) {
+            throw new IllegalArgumentException("Only GRPC channels are allowed for emulator.");
```

Comment:
We use the emulator in Spring Cloud GCP with REST. Maybe [this](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/11c070e2ba0bc17b5ba628ed763bb8d0144ade5a/spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java#L93) could be helpful?